engine/execution/rpc/engine.go

```diff
@@ -394,6 +422,15 @@ func (h *handler) GetTransactionResultsByBlockID(
 		return nil, status.Errorf(codes.InvalidArgument, "invalid blockID: %v", err)
 	}
 
+	// must verify block exists first, since transactionResults.ByBlockID will return an empty slice
+	// if block does not exist
+	if _, err = h.headers.ByBlockID(blockID); err != nil {
```

Comment:
we could check if the block has been executed, instead of exists or not
```suggestion
	if _, err = h.commits.ByBlockID(blockID); err != nil {
```